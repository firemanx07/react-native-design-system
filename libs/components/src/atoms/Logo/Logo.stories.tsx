import { ColorType } from '@proxym/themes';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import {
  StorybookRow,
  StorybookScreen,
  StorybookSection,
  StorybookSectionText,
} from '../../storybook';
import { Logo as LogoBase, PropsType as LogoProps } from './Logo';

const Story: ComponentMeta<typeof LogoBase> = {
  component: LogoBase,
  title: 'Atoms/Logo',
  argTypes: {
    color: {
      control: { type: 'color' },
    },
  },
};
export default Story;

const defaultArgs: LogoProps = {
  color: '#222222',
  width: 131,
};

const Template: ComponentStory<typeof LogoBase> = (args: LogoProps) => (
  <StorybookScreen>
    <StorybookSection>
      <LogoBase {...args} />
    </StorybookSection>
    <StorybookSectionText>Sample variants:</StorybookSectionText>
    <StorybookRow>
      <StorybookSection title="Color primary">
        <LogoBase {...defaultArgs} color={ColorType.primary} />
      </StorybookSection>
      <StorybookSection title="Color dark">
        <LogoBase {...defaultArgs} color={ColorType.dark} />
      </StorybookSection>
      <StorybookSection title="Color light">
        <LogoBase {...defaultArgs} color={ColorType.light} />
      </StorybookSection>
    </StorybookRow>
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['color', 'width'],
  },
};

export const Logo = Template.bind({});
Logo.parameters = parameters;
Logo.args = {
  ...defaultArgs,
};
