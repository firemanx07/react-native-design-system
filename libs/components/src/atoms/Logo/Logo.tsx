import { useNamespacedTheme } from '@proxym/themes';
import React, { memo } from 'react';

import CelebrateLogo from '../../assets/images/logo.svg';
import { TestIDType } from '../../types';

export type PropsType = {
  color?: string;
  width?: number;
} & TestIDType;

const BASE_WIDTH = 131;
const BASE_HEIGHT = 36;

export const Logo = ({ width = BASE_WIDTH, color, testID }: PropsType) => {
  const height = (width * BASE_HEIGHT) / BASE_WIDTH;
  const { colors } = useNamespacedTheme();

  return (
    <CelebrateLogo
      width={width}
      height={height}
      fill={color || colors.primary}
      testID={testID}
    />
  );
};

export default memo(Logo);
