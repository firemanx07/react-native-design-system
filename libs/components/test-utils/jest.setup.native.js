import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
