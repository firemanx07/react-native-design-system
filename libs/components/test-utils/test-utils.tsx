import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
