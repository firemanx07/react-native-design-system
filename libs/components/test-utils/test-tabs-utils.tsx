import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
