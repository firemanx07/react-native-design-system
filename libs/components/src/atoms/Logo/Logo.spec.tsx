import { testAllStories } from '../../../test-utils';
import * as stories from './Logo.stories';

describe('Logo tests', () => {
  testAllStories(stories);
});
