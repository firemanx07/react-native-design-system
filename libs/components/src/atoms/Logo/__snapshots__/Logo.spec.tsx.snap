// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo tests Logo snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "backgroundColor": "#DEDFE0",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <test-file-stub
      fill="#2C2C2C"
      height={36}
      width={131}
    />
  </View>
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "color": "#2C2C2C",
          "marginBottom": 4,
        },
        {
          "color": "#2C2C2C",
        },
      ]
    }
  >
    Sample variants:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Color primary
      </Text>
      <test-file-stub
        fill="#4D37D9"
        height={36}
        width={131}
      />
    </View>
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Color dark
      </Text>
      <test-file-stub
        fill="#2C2C2C"
        height={36}
        width={131}
      />
    </View>
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Color light
      </Text>
      <test-file-stub
        fill="#FFFFFF"
        height={36}
        width={131}
      />
    </View>
  </View>
</View>
`;
